### ISSUE TITLE
How to display relation of adopted children

### ISSUE THREAD

**Issue by schuco** [Neriderc/GVExport#132]
I have some relatives in my tree which have been entered already with PAF long before present GEDCOM standard.
One of these is  Elisabeth Hoerdemann, the biological daughter of Heinrich Anton H. and Theodora Clementine Dieckhöfer was adopted by the widow of family Heinrich Welkener and Marianne Hoerdemann.
With GVE2.1.7 they were displayed with arrows between bilological as well as adopting family. Correctly the adopting family is shown non-related:
![ADOP old](https://user-images.githubusercontent.com/5896925/184497764-cad272d3-20a3-4140-8202-5c9a1bb01c8c.png)

Today I checked for errors in my tree by the utility given in webtrees and among many others the following error was listed:
_Familie: [F269](https://brandneu.schulte-coerne.de/index.php?route=%2Ftree%2Ftree2%2Frecord%2FF269) — Zeilennummer: 11 — 2 ADOP 
Ungültiges GEDCOM-Kennzeichen. FAM:CHIL:ADOP_

Then I looked up in the GEDCOM 5.5.1 Documentation and found this:
**Adoption Events:
In GEDCOM 5.x, the ADOPtion event was moved from the FAM.CHIL structure to the
INDI.FAMC.ADOP structure, it also appears in the INDI.ADOP structure. In GEDCOM 5.4 the
ADOPtion event appears only as an individual event which optionally contains a FAMC pointer to the
adoptive family. Subordinate to this pointer is another ADOPtion tag which indicates whether the
HUSB or WIFE in the pointed at family was the adoptive parent**

Following this I deleted 
1 CHIL @I461@
2 ADOP
in the Family-record of the adopting parents.

What I get now from GVE is:
![ADOP new](https://user-images.githubusercontent.com/5896925/184497970-c5061db3-e54f-4135-847b-8cf4de0cd077.png)

In this small diagram you may be able to recognize the special family structure, but within a bigger diagram you would never recognize a relation between adopting family and child.
I know, that an arrow between adopting family and child is not quite correct as it mixes biological and adopted relations. It is desirable to have a distinctly structured or colored arrow for that.
But as long as this is not implemented in GVE I would prefer to see an arrow in adoption relations too. This seems a better compromise than not having this, as it is quite easy to detect adopted children being always pointed to by two arrows.

So I have 3 questions:
_1. Is it intended in present GVE to have no arrow pointing from the adopting family to the adopted child?
2. Is it indended that incorrect use of the GEDCOM 5.5.1-standard leads to an alternative display?
3. Will the incorrect FAM.CHIL structure (which presently helps me to get the desired diagram) become a problem with future webtree versions under GEDCOM 7?_ 

A somewhat related question, which @hartenthaler may be able to answer:
I see that in my tree the relationship coefficient in the vesta module is wrongly including adopted relations. Is this due to my non-standard practice to code adoptions?

**Comment by Neriderc** [comment]
The lack of arrow is not intended. In your first screen shot, where both families have arrows and the adoptive parents are shown as non-related, this is how it should be shown. We have a later plan to display different colour arrows.

Just to confirm, it works ok in GVE2.1.7 but not in GVE2.1.7.2? Or have I misunderstood this?

**Comment by schuco** [comment]
Both are from GVE 2.1.7.2 (master).
The difference is from my Gedcom file
1. correct arrows but incorrect Gedcom (ADOP in FAM.CHILD entry)
2. Missing arrow with formally correct Gedcom. (No child entry in adopting family)

**Comment by Neriderc** [comment]
Ah sorry I missed that it was you correcting the data that caused the change. I will have a go at setting up this scenario for testing, and see where the issue lies. Theres a good chance this would have been an issue regardless of GVE version, but I will have a play and see if I can recreate it, then will see what it takes to fix it.

**Comment by hartenthaler** [comment]
> I see that in my tree the relationship coefficient in the vesta module is wrongly including adopted relations. Is this due to my non-standard practice to code adoptions?

I will check this.

**Comment by Neriderc** [comment]
@schuco I think I don't have the knowledge of GEDCOM that you have, I haven't been able to recreate the issue you describe.

When I use webtrees to create the adoptive link, what I get is the child listed but the mention of adoption is not in the family record. Perhaps when you deleted this:
> 1 CHIL @i461@
> 2 ADOP

Maybe you should have only deleted the second line? Honestly, I'm not sure if the issue is your GEDCOM or whether it's that GVE expects it a certain way but other valid values exist.

Once you removed this record, was there any reference to the child in the family GEDCOM? If not, that may be why the arrow doesn't show. But whether or not that is valid I'm not sure. Maybe @hartenthaler knows more about what is valid?

For detecting adoptions, I reactivated old code and made modifications to make it work with the up to date webtrees, I haven't made any decisions myself on how to identify the adoptions in the GEDCOM, so it's perfectly possible it's out of date.

**Comment by schuco** [comment]
> `Perhaps when you deleted this:
> 
>     1 CHIL @i461@
>     2 ADOP
> Maybe you should have only deleted the second line? `

In case 2 I deleted both lines inclusive 1 CHIL
I am not anGedcom expert, just the opposite. But as I understand the documentation cited, this had to be done to be in accordance with 5.5.1 standard.
I recall my comment on the utility in webtreesto detect errors in a tree. I had no errors there under wt2.0. Now I see lots of them  including the ADOP issue.

I will be away next week so I may not be able to respond quickly.

**Comment by Neriderc** [comment]
When you made this change, does webtrees detect any errors with this record?

I'm happy to to look at resolving the issue, but first I'd like to confirm that your GEDCOM is valid, and I don't have the knowledge to confirm this. Perhaps the webtrees forum is the right place for asking for help on making sure the GEDCOM record is set up correctly?

**Comment by hartenthaler** [comment]
I will check the GEDCOM standards 5.5.1 and 7.0 regarding adopted children in the next few days and will develop a test scenario.

**Comment by schuco** [comment]
I have looked into the issue again, that GVE does not show arrows between adopting parents and adopted child in some cases (presented in my second screenshot above).
I now found that **GVE handles adoption properly with arrows if the GEDCOM for adoption is set up properly**.

**The error has been on my side:**
Error-checking my GEDCOM-file by the built-in utility in webtrees I got following error:
_Familie: [F269](https://brandneu.schulte-coerne.de/index.php?route=%2Ftree%2Ftree2%2Frecord%2FF269) — Zeilennummer: 11 — 2 ADOP
Ungültiges GEDCOM-Kennzeichen. FAM:CHIL:ADOP_

The corresponding family record in the GEDCOM was:
_0 @F197@ FAM
1 CHIL @I1159@
2 ADOP_

What I wrongly did was deleting both, the 1 CHIL and the 2 ADOP lines. 
When I correctly would have left the 1 CHIL line the arrow would have been displayed correctly.

So in my view the issue can be closed in the GVE context.
